src/ansys/simai/core/utils/configuration.py

```diff
@@ -25,47 +25,75 @@
 from typing import Optional
 from urllib.parse import urlparse, urlunparse
 
-from pydantic import AnyHttpUrl, BaseModel, Field, HttpUrl, root_validator, validator
+from pydantic import (
+    AnyHttpUrl,
+    BaseModel,
+    Field,
+    HttpUrl,
+    field_validator,
+    model_validator,
+    validator,
+)
+from pydantic_core import PydanticCustomError
 
 from ansys.simai.core.utils.misc import prompt_for_input
 
 logger = logging.getLogger(__name__)
 
 
-def prompt_for_input_factory(*args, **kwargs):
-    return lambda: prompt_for_input(*args, **kwargs)
+def prompt_if_interactive(interactive, **kwargs):
+    """Raise an error or prompt for input according to _interactive_mode."""
+    if not interactive:
+        raise PydanticCustomError(
+            "conf_param_missing", f"""Missing parameter "{kwargs["name"]}" from configuration"""
+        )
+    return prompt_for_input(**kwargs)
 
 
 class Credentials(BaseModel, extra="forbid"):
-    username: str = ""  # dummy default, the root validator will call prompt_for_input
+    username: str  # the root validator will call prompt_for_input
     "Username: Required if :code:`Credentials` is defined, automatically prompted."
-    password: str = ""  # dummy default, like above
+    password: str  # like above
```

Comment:
While these comments are now correct, the original comments were there to explain a code oddity that is now gone. So I don't see why they are needed anymore.
Especially since "totp" and "organization" don't have the same comment.

Reply:
Comments now reflect the functionality of `Credentials`, functionality for organization `organization` is explained in its validator `check_organization_exists`